File: board-front/src/types/post.ts
export interface Post {
  id: number
  title: string
  content: string
  createdAt: string
  viewCount: number
  likeCount: number
  userId: number | null
}

// 投稿作成時に送信するデータの型
export interface CreatePostRequest {
  title: string
  content: string
}

// ページング付きレスポンスの型
export interface PagedPosts {
  items: Post[]
  totalCount: number
  page: number
  pageSize: number
}

File: BoardApi/Models/Post.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace BoardApi.Models
{
    public class Post
    {
        public int Id { get; set; }

        [Required]
        [MaxLength(200)]
        public string Title { get; set; }

        [Required]
        public string Content { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public int ViewCount { get; set; } = 0;

        public int LikeCount { get; set; } = 0;

        public bool DelFlag { get; set; } = false;

        public int? UserId { get; set; }
        public User User { get; set; }

        public ICollection<Comment> Comments { get; set; }
    }
}
